client/src/components/HostMenu.jsx

```diff
@@ -18,12 +18,27 @@ const HostMenu = ({ guests, menu, setTopLevelState, sessionId }) => {
     socket.emit('hostJoined', { sessionId });
   }, []);
 
+  useEffect(() => {
+    setCurrentName(guests[0].guestName);
+  }, []);
+
   const onChange = (event) => {
     setCurrentName(event.target.value);
     onCloseClick();
     event.preventDefault();
   };
 
+  const onModalClick = (event) => {
+    const modal = document.getElementById('myModal');
+    modal.style.display = 'block';
+    event.preventDefault();
+  };
```

Comment:
Don't do this now, but React doesn't like when you scan the DOM. It prefers using `useRef` to access DOM elements.
```suggestion
  // somewhere up at the top
  const modalRef = useRef(null);
  
  const onModalClick = (event) => {
    modalRef.current.style.display = 'block';
    event.preventDefault();
  };
  
  // wherever the modal is
  <Modal ref={modalRef} />
```